client/my-sites/plan-features-2023-grid/hooks/npm-ready/use-is-large-currency.ts

```diff
@@ -15,12 +16,18 @@ interface Props {
 export default function useIsLargeCurrency( { gridPlans, returnMonthly = true }: Props ) {
 	return gridPlans.some( ( gridPlan ) => {
 		const {
-			pricing: { originalPrice, discountedPrice },
+			pricing: { originalPrice, discountedPrice, currencyCode },
 		} = gridPlan;
 
 		return [
 			originalPrice[ returnMonthly ? 'monthly' : 'full' ],
 			discountedPrice[ returnMonthly ? 'monthly' : 'full' ],
-		].some( ( price ) => price && price.toString().length > LARGE_CURRENCY_CHAR_THRESHOLD );
+		].some( ( price ) => {
+			const formattedPrice = formatCurrency( price || 0, currencyCode || '', {
```

Comment:
**Note:**

@sirbrillig This hook determines if the number of numerical digits for a display price are too long to represent properly in the plans pricing grid. Since we’re trying to switch over to use smallest units, do you have happen to have suggestions on how to best account for this ( since $20 is now represented as the integer 2000, instead of 20 )? I thought about using `formatCurrency` or `getCurrencyObject` to account for smallest units, but the return values from those functions are display strings that include currency symbols and place separators ( periods, commas, etc. ).

I’m wondering if this is a valid use case to modify the interface of the `format-currency` library to provide the necessary info. Alternatively, I might be missing something, so I’d love your thoughts

Reply:
🤔 I have two questions:

1. Is there any way to use CSS for this sort of display logic rather than code? For example, it looks like [here](https://github.com/Automattic/wp-calypso/blob/4d8d10c2b6ae92e0fec9d2ec679bcad9f326dc3c/client/my-sites/plan-features-2023-grid/components/actions.tsx#L88-L103) a label without the price is displayed if the currency is "large". Another option might be to render both the "with price" and "without price" versions and hide one using a [css container query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries) unless the container is large enough to display the price? 
2. If the concern is sufficient space to display the formatted price, don't you want to know the length of the price as a string with formatting? The price `100.50 HK$` is 5 numbers (6 if you count the decimal) but takes up 10 characters, whereas the price `¥505001` also has 6 numbers but takes up only 7 characters. Some currencies have relatively long currency symbols like GEL which uses `Lari` or VEF which uses `Bs. F.`. 

Also worth noting that not all characters are the same width (especially in some languages), which makes using a "number of digits" for display fit purposes somewhat dubious.

Reply:
Thanks for the insight Payton! 

> Is there any way to use CSS for this sort of display logic rather than code? For example, it looks like [here](https://github.com/Automattic/wp-calypso/blob/4d8d10c2b6ae92e0fec9d2ec679bcad9f326dc3c/client/my-sites/plan-features-2023-grid/components/actions.tsx#L88-L103) a label without the price is displayed if the currency is "large". Another option might be to render both the "with price" and "without price" versions and hide one using a [css container query](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_container_queries) unless the container is large enough to display the price?

@aneeshd16, I believe you worked on this? I'm wondering if these approaches were considered and if there's any hidden gotchas to using CSS containers?

> If the concern is sufficient space to display the formatted price, don't you want to know the length of the price as a string with formatting? The price 100.50 HK$ is 5 numbers (6 if you count the decimal) but takes up 10 characters, whereas the price ¥505001 also has 6 numbers but takes up only 7 characters. Some currencies have relatively long currency symbols like GEL which uses Lari or VEF which uses Bs. F..

Your point about relatively long currency symbols and your example with HK$ makes sense to me. 

My best guess is that perhaps the smaller formatting and font-sizes for currency symbols and separator symbols might make their widths relatively negligible. @aneeshd16 wondering if you have thoughts here as well?

Reply:
If you don't care about the currency symbol, one other idea as an interim fix might be to use `getCurrencyObject()`:

```
const currencyObject = getCurrencyObject( /* ... */ );
const priceToMeasure = currencyObject.integer + currencyObjecy.hasNonZeroFraction ? currencyObject.fraction : '';`
const priceLength = priceToMeasure.length;
```

That way you are including all the other price formatting (commas, spaces, and periods) in the measurement.

Reply:
Made the changes in https://github.com/Automattic/wp-calypso/pull/81537/commits/632eaaa8696c7015aa97100eb925f03db8b8d885

Reply:
Hi, thanks for the ping!

> I believe you worked on this? I'm wondering if these approaches were considered and if there's any hidden gotchas to using CSS containers?

The challenge was that all prices in the pricing grid should be displayed in the same font size. There is a possibility that the price of one plan exceeds the threshold while others don't. For example, in certain currencies, the price of the eCommerce plan may be classified as "large" based on our heuristic, while the price of the free plan (0) will not be "large". This is why we cannot use CSS solutions that act on a single element, as we need to consider all currencies displayed on the grid. This is also why this hook accepts a list of plans instead of a single plan.

Since we need to consider currency symbols and space/decimal/comma/etc characters as well, why don't we check for the length of the string produced by `formatCurrency`? This would have the same result as the current implementation.

```js
const formattedPrice = formatCurrency( price || 0, currencyCode || '', {
  stripZeros: true,
  isSmallestUnit: true,
} );
return formattedPrice.length > LARGE_CURRENCY_CHAR_THRESHOLD;
```

> I've checked every single currency we support from USD to TRY. Everything looks reasonable if we adjust the threshold from 5 to 6 -- happy to provide screenshots

I don't see anything wrong with this approach either since `LARGE_CURRENCY_CHAR_THRESHOLD` was also based on observation and not on any other rule. 👍